File: domain/recruit.go
package domain

import "time"

type RecruitKind int

const (
	RecruitKindInternship RecruitKind = iota
	RecruitKindHackathon
	RecruitKindEvent
	RecruitKindJob
	RecruitKindCtf
	RecruitKindWorkshop
)

type Recruit struct {
	ID               uint      `gorm:"primary_key"`
	Name             string    `gorm:"not null; type:varchar(100)"`
	RecruitStartDate time.Time `gorm:"type:timestamp"`
	RecruitEndDate   time.Time `gorm:"type:timestamp"`
	Technologies     []string
	Link             string
	Kind             RecruitKind
	Company          string
	Area             []string
	Reward           string
	Schedule1        string
	Schedule2        string
	Schedule3        string
	Comment          string
}

type RecruitRepo interface {
	List() ([]Recruit, error)
	Create(*Recruit) error
}

File: src/recruit.rb
class Recruit
  attr_accessor :title, :recruit_start_date, :recruit_end_date, :technologies, :link, :kind, :company, :area, :reward, :schedule1, :schedule2, :schedule3, :comment

  def initialize(**params)
    params.each{ |k, v| send("#{k}=", v) if self.methods.include?(k) }
  end

  def to_h
    self.instance_variables.each_with_object({}) do |var, hash|
      hash[var.to_s.delete('@')] = self.instance_variable_get(var)
    end
  end
end
